source/src/plugins/MeanWithinExpectedTest.cc

```diff
@@ -110,20 +122,64 @@ namespace dqm4hep {
         throw StatusCodeException(STATUS_CODE_INVALID_PTR);
       }
 
-      const float mean(pHistogram->GetMean());
-      const float range(fabs(m_meanDeviationUpper - m_meanDeviationLower));
+      float result;
+
+      // Worth remembering that this is useless right now for multiple test-types; there's only one set of
+      // parameters and this will only allow the last test type to be operated on by the rest of the code
 
-      if (m_meanDeviationLower < mean && mean < m_meanDeviationUpper) {
-        report.m_message =
-            "Within expected range: expected " + typeToString(m_expectedMean) + ", got " + typeToString(mean);
-      } 
+      // It's not relevant for now, since we're unable to do more than one test type per "qtest" object,
+      // since the tests only operate on one set of parameters.
+
+      for (std::vector<std::string>::iterator it = m_testTypes.begin(); it != m_testTypes.end(); it++){
+	result = AnalysisHelper::mainHelper(monitorElement, *it, m_percentage);
```

Comment:
You are not testing all tests here, but only the last one. The value of `result` is replaced at every iteration